host_core/lib/host_core/host.ex

```diff
@@ -68,9 +68,95 @@ defmodule HostCore.Host do
     {:reply, state.host_key, state}
   end
 
-  defp start_gnat() do
-    {:ok, _gnat} = Gnat.start_link(%{host: '127.0.0.1', port: 4222}, name: :lattice_nats)
-    {:ok, _gnaT} = Gnat.start_link(%{host: '127.0.0.1', port: 4222}, name: :control_nats)
+  defp start_gnat(opts) do
+    configure_lattice_gnat(%{
+      host: opts.rpc_host,
+      port: opts.rpc_port,
+      username: opts.rpc_user,
+      password: opts.rpc_pass,
+      token: opts.rpc_token,
+      nkey_seed: opts.rpc_seed,
+      jwt: opts.rpc_jwt
+    })
+
+    configure_control_gnat(%{
+      host: opts.ctl_host,
+      port: opts.ctl_port,
+      username: opts.ctl_user,
+      password: opts.ctl_pass,
+      token: opts.ctl_token,
+      nkey_seed: opts.ctl_seed,
+      jwt: opts.ctl_jwt
+    })
+  end
+
+  defp configure_lattice_gnat(opts) do
+    conn_settings =
+      Map.merge(%{host: opts.host, port: opts.port}, determine_auth_method(opts, "lattice"))
+
+    case Gnat.start_link(conn_settings, name: :lattice_nats) do
+      {:ok, _gnat} ->
+        :ok
+
+      {:error, :econnrefused} ->
+        Logger.error("Unable to establish lattice NATS connection, connection refused")
+
+      {:error, _} ->
+        Logger.error("Authentication to lattice NATS connection failed")
+    end
+  end
+
+  defp configure_control_gnat(opts) do
+    conn_settings =
+      Map.merge(
+        %{host: opts.host, port: opts.port},
+        determine_auth_method(opts, "control interface")
+      )
+
+    case Gnat.start_link(conn_settings, name: :control_nats) do
+      {:ok, _gnat} ->
+        :ok
+
+      {:error, :econnrefused} ->
+        Logger.error("Unable to establish control interface NATS connection, connection refused")
+
+      {:error, _} ->
+        Logger.error("Authentication to control interface NATS connection failed")
+    end
+  end
+
+  defp determine_auth_method(
+         %{
+           username: username,
+           password: password,
+           token: token,
+           nkey_seed: nkey_seed,
+           jwt: jwt
+         },
+         conn_name
+       ) do
+    cond do
+      jwt != "" && nkey_seed != "" ->
+        Logger.info("Authenticating to #{conn_name} NATS with JWT and seed")
+        %{jwt: jwt, nkey_seed: nkey_seed, auth_required: true}
+
+      nkey_seed != "" ->
```

Comment:
This looks like an odd case. If we have a seed but we do _not_ have a JWT (what this predicate indicates), then we will get an authentication failure. User seed and JWT must always exist at the same time and we can't have one without the other.

Reply:
Oh gotcha, I was basing my cases off of the Gnat documentation here: https://hexdocs.pm/gnat/readme.html#authentication

You can see there's a case where just an nkey is accepted. If that's always going to be invalid, we can drop an error log here to handle that (or just remove the case entirely)

Reply:
I stand corrected. It is in fact, entirely possible to authenticate using nothing but a user NKey and a challenge, as per [this section](https://docs.nats.io/nats-server/configuration/securing_nats/auth_intro/nkey_auth#generating-nkeys-and-configuring-the-server) of the NATS docs.

Carry on. As you were.